kv/dist_sender.go

```diff
@@ -184,6 +186,10 @@ func NewDistSender(ctx *DistSenderContext, gossip *gossip.Gossip) *DistSender {
 	if ctx.RangeLookupMaxRanges <= 0 {
 		ds.rangeLookupMaxRanges = defaultRangeLookupMaxRanges
 	}
+	ds.tracer = ctx.Tracer
+	if ds.tracer == nil {
+		ds.tracer = util.NewTracer(nil, 2)
```

Comment:
This is going away, I added this before I patched up the Feed. The intention is that this can be nil as you suggest.